Add target* params for conflict targeting (#266)
spotify/missinglink: README.md

By default, all subpackages of the specified packages are also ignored, but
this can be disabled on an individual basis by adding

> `<filterSubpackages>false</filterSubpackages>` to the `<ignoreSourcePackage>`

Reviewer: maybe worth adding a brief note below this that the param used to be called `ignoreSubpackages`, its been renamed, but old name is supported as well

... and I just remembered this repo has a CHANGELOG.md that would be good to update too, both about the new functionality and this change

Author: great idea! updated both

or `<ignoreDestinationPackage>` element.

### Target only conflicts from certain packages

> Conversely, found conflicts can be filtered based on the package name of the class that has the

Reviewer: a small suggestion that I think might make this read easier: 

```suggestion
Conversely, the plugin can be configured to _only_ report on conflicts in specific packages, based on the name of the class that has the
```

(I think this messes up the word wrapping though)
```

conflict. There are separate configuration options for targeting conflicts on the "source" side of the conflict
and the "destination" side of the conflict.

Packages on the source side can be targeted with `<targetSourcePackages>` and
packages on the destination side can be targeted with `<targetDestinationPackages>`:

```xml
<configuration>
  <!-- Only target conflicts coming from groovy.lang source package -->
  <targetSourcePackages>
    <targetSourcePackage>
      <package>groovy.lang</package>
    </targetSourcePackage>
  </targetSourcePackages>
  <!-- Only target conflicts coming from com.foo package on the callee side -->
  <targetDestinationPackages>
    <targetDestinationPackage>
      <package>com.foo</package>
    </targetDestinationPackage>
  </targetDestinationPackages>
</configuration>
```

By default, all subpackages of the specified packages are also ignored, but
this can be disabled on an individual basis by adding
`<filterSubpackages>false</filterSubpackages>` to the `<ignoreSourcePackage>`
or `<ignoreDestinationPackage>` element.

> Note that `allow*` options CANNOT be used in conjunction with `ignore*` options. You can only specify one or the other.

Reviewer: ```suggestion
**Note** that `target*` options CANNOT be used in conjunction with `ignore*` options. You can only specify one or the other.
```

> Note that `target*` options CANNOT be used in conjunction with `ignore*` options. You can only specify one or the other.

Reviewer: just to add some emphasis

```suggestion
**Note** that `target*` options CANNOT be used in conjunction with `ignore*` options. You can only specify one or the other.
```

By default, all subpackages of the specified packages are also ignored, but
this can be disabled on an individual basis by adding
`<filterSubpackages>false</filterSubpackages>` to the `<ignoreSourcePackage>`
or `<ignoreDestinationPackage>` element.

> ### Allow only conflicts from certain packages

Reviewer: Maybe this is a bit meta, but if I am understanding right it is not really that you want to "allow" conflicts in these packages, but rather this would be used to make missinglink report an error _only_ when the conflict happens in these packages, is that right? 

I wonder if using "allow" in the description and the property name would be confusing then, but at the same time I'm not sure what a better term would be - maybe something like "failOnConflictInPackages" or "target packages", what do you think?

Or maybe something along the lines of "packages to look for conflicts in" - its hard to think of a single term that describes this (unless I am just having a brain freeze)

Author: Rename to `target*`. lmk what you think!

Conversely, conflicts can be allowed based on the package name of the class that has the
conflict. There are separate configuration options for allowing conflicts on
the "source" side of the conflict and the "destination" side of the conflict.

> Packages on the source side can be allowe with `<allowSourcePackages>` and

Reviewer: ```suggestion
Packages on the source side can be allowed with `<allowSourcePackages>` and
```
